apps/emqx_connector_aggregator/src/emqx_connector_aggregator.erl

```diff
@@ -186,10 +191,10 @@ init(St0 = #st{name = Name}) ->
 
 handle_call({next_buffer, Timestamp}, _From, St0) ->
     St = #st{buffer = Buffer} = handle_next_buffer(Timestamp, St0),
-    {reply, Buffer, St, 0};
+    {reply, Buffer, St, {continue, enqueue_delivery}};
```

Comment:
If I remember the original idea correctly, this 0 timeout is here to serve 2 purposes. The assumption is that if there's a `{next_buffer, ...}` call, high chance that there are actually _a lot_ of concurrent `{next_buffer, ...}` calls (same with `{rotate_buffer, ...}`, etc.).
1. If there are, make sure that all of them are processed and replied to first, only then start delivery (which is moderately expensive).
2. At the same time, do not overwhelm the current process with a lot of `enqueue_delivery` requests.

However you're absolutely right that this mechanism is way too brittle.

That said, what do you think about doing `gen_server:cast/2` to `self()` in the first clause of `enqueue_closed_buffer/2`? This IIUC should mostly preserve the current mechanics.

Reply:
Sure, I think that achieves the same goal.  I've pushed a patch for that.